proto/data.proto

```diff
@@ -133,6 +133,12 @@ message ChangeReplicasTrigger {
   repeated Replica updated_replicas = 4 [(gogoproto.nullable) = false];
 }
 
+// CommitTrigger encapsulates all of the internal-only commit triggers.
+message InternalCommitTrigger {
+  optional SplitTrigger split_trigger = 1;
```

Comment:
Can you use the new `oneof` functionality here? Or is that not supported in gogoproto?